pkg/configurator/validating_webhook.go

```diff
@@ -0,0 +1,348 @@
+package configurator
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+	"time"
+
+	"k8s.io/api/admission/v1beta1"
+	admissionv1beta1 "k8s.io/api/admissionregistration/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/openservicemesh/osm/pkg/certificate"
+	"github.com/openservicemesh/osm/pkg/constants"
+	"github.com/openservicemesh/osm/pkg/webhook"
+)
+
+var (
+	codecs       = serializer.NewCodecFactory(runtime.NewScheme())
+	deserializer = codecs.UniversalDeserializer()
+
+	// boolFields are the fields in osm-config that take in a boolean
+	boolFields = []string{"egress", "enable_debug_server", "permissive_traffic_policy_mode", "prometheus_scraping", "tracing_enable", "use_https_ingress"}
+
+	// ValidEnvoyLogLevels is a list of envoy log levels
+	ValidEnvoyLogLevels = []string{"trace", "debug", "info", "warning", "warn", "error", "critical", "off"}
+
+	// defaultFields are the default fields in osm-config
+	defaultFields = []string{"egress", "enable_debug_server", "permissive_traffic_policy_mode", "prometheus_scraping", "tracing_enable", "use_https_ingress", "envoy_log_level", "service_cert_validity_duration", "tracing_address", "tracing_port", "tracing_endpoint"}
+)
+
+const (
+	// ValidatingWebhookName is the name of the validating webhook used for validating osm-config
+	ValidatingWebhookName = "validating-webhook.k8s.io"
```

Comment:
Can we change the name throughout to `osm-config-webhook.k8s.io` to indicate this webhook is for the osm-config configMap

Reply:
sure thing!